### PR TITLE
Eric/compact

### PR DESCRIPTION
**Pull request by EricBAndrews**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

## About this Pull Request
By overwhelming popular demand, re-enabled compact mode, and gave it a little facelift to align with our post flow.

- There are now three post sizes, compact, headline, and large
- Settings has been updated accordingly; `shouldShowCompactPosts: Bool` is now `postSize: Enum`
- Updated quick-switcher in the toolbar

This PR also consumes #186.

## Screenshots and Videos
all compact

new menu

## Additional context
It's a little dirty right now--there isn't great consistency between display ("compact/headline" and backend "ultracompact/compact") and there's some code reuse possible, but this is a "get it in because people are mad" kinda PR--I've opened an issue to pretty it up

### CODE REVIEW

File: Mlem/Views/Shared/Cached Image With Nsfw Filter.swift [hunk 1]
```diff
@@ -35,6 +35,7 @@ struct CachedImageWithNsfwFilter: View {
                     .clipShape(RoundedRectangle(cornerRadius: 8))
                     .overlay(RoundedRectangle(cornerRadius: 8)
                         .stroke(Color(UIColor.secondarySystemBackground), lineWidth: 1))
+                    .allowsHitTesting(false)
```
Comment (JakeShirley) [inline Mlem/Views/Shared/Cached Image With Nsfw Filter.swift:38]: Is this a related change? Or is this your bugfix around clicking image posts?

**Review by JakeShirley — APPROVED** [review]
I love the concept here.  Outside the scope of this story but I still think that previewing the post in settings would be awesome for honing in on settings users want.

**Review by WestonHanners — APPROVED** [review]
